packages/compass-import-export/src/export/export-json.spec.ts

```diff
@@ -0,0 +1,455 @@
+                $lte: 2,
+              },
+            },
+          },
+          {
+            $project: {
+              name: 1,
+              doubleName: '$name',
+              _id: 0,
+            },
+          },
+        ],
+        options: {},
+      },
+      output,
+      variant: 'default',
+      abortSignal: abortController.signal,
+      progressCallback: () => {},
+    });
+
+    expect(result.docsWritten).to.equal(3);
+    expect(result.aborted).to.be.false;
+
+    let resultText;
+    let writtenResultDocs;
+    try {
+      resultText = await fs.promises.readFile(resultPath, 'utf8');
+      writtenResultDocs = EJSON.parse(resultText);
+    } catch (err) {
+      console.log(resultPath);
+      throw err;
+    }
+
+    const expectedText = `[{
+  "name": "pineapple",
+  "doubleName": "pineapple"
+},
+{
+  "name": "apple",
+  "doubleName": "apple"
+},
+{
+  "name": "orange",
+  "doubleName": "orange"
+}]`;
+
+    expect(writtenResultDocs).to.deep.equal(
+      ['pineapple', 'apple', 'orange'].map((name) => ({
+        name,
+        doubleName: name,
+      }))
+    );
+    expect(resultText).to.deep.equal(expectedText);
+  });
+
+  it('handles an empty collection', async function () {
+    const abortController = new AbortController();
+
+    const resultPath = path.join(tmpdir, 'test-empty.exported.ejson');
+
+    const output = fs.createWriteStream(resultPath);
+    const result = await exportJSON({
+      dataService,
+      ns: `${testDB}.test-empty`,
+      output,
+      variant: 'default',
+      abortSignal: abortController.signal,
+      progressCallback: () => {},
+    });
+
+    expect(result.docsWritten).to.equal(0);
+    expect(result.aborted).to.be.false;
+
+    let resultText;
+    let writtenResultDocs;
+    try {
+      resultText = await fs.promises.readFile(resultPath, 'utf8');
+      writtenResultDocs = EJSON.parse(resultText);
+    } catch (err) {
+      console.log(resultPath);
+      throw err;
+    }
+
+    const expectedText = '[]';
+
+    expect(writtenResultDocs).to.deep.equal([]);
+    expect(resultText).to.deep.equal(expectedText);
+  });
+
+  it('throws when the database stream errors', async function () {
+    const abortController = new AbortController();
+
+    const mockReadStream = new Readable({
+      objectMode: true,
+      read: function () {
+        this.emit('error', new Error('example error cannot fetch docs'));
+      },
+    });
+
+    sinon.stub(dataService, 'findCursor').returns({
+      stream: () => mockReadStream,
+    } as unknown as FindCursor);
+
+    try {
+      await exportJSON({
+        dataService,
+        ns: testNS,
+        output: temp.createWriteStream(),
+        variant: 'default',
+        abortSignal: abortController.signal,
+        progressCallback: () => {},
+      });
+
+      throw new Error('did not expect export to succeed');
+    } catch (err: any) {
+      expect(err.message).to.equal('example error cannot fetch docs');
```

Comment:
if you look at some of my tests you can:

```typescript
    await expect(promise).to.be.rejectedWith(
      Error, // class of error
      'Parser cannot parse input: expected a value' // message
    );
```

That way you can make sure the code throws whereas your tests now would pass if the code somehow succeeds. a bit neater than the try/catch as well.

So it would be something like:

```typescript
await expect(exportJSON({
        dataService,
        ns: testNS,
        output: temp.createWriteStream(),
        variant: 'default',
        abortSignal: abortController.signal,
        progressCallback: () => {},
      }).to.reject(Error, 'example error cannot fetch docs');
```

It would also give you better test failure output. ie. it would print what it _did_ get.

Reply:
Ah this is good. Thanks for mentioning. Updated tests.